api/src/main/java/ai/djl/ndarray/NDArray.java

```diff
@@ -4625,4 +4625,30 @@ default NDArray oneHot(int depth) {
      *     href=https://d2l.djl.ai/chapter_linear-networks/softmax-regression.html#classification-problems>Classification-problems</a>
      */
     NDArray oneHot(int depth, float onValue, float offValue, DataType dataType);
+
+    /**
+     * Batchwise product of this {@code NDArray} and the other {@code NDArray}.
+     *
+     * <ul>
+     *   <li> batch_dot is used to compute dot product of x and y when x and y are data in batch, namely N-D (N >= 3) arrays in shape of (B0, …, B_i, :, :).
```

Comment:
You got this from MXNet and there should be a next sentence so include that information as well. Also, feel free to edit this description for grammar, clarity, comprehendability, etc. Don't feel like you are beholden to their exact words. You can even write the description on your own if you prefer

Reply:
Cool thanks. Yeah for now I will just stick with this documentation but I will be sure to change it for any future additions if I consider it necessary.